js/modules/k6/http/http.go

```diff
@@ -240,6 +257,60 @@ func (*HTTP) request(ctx context.Context, rt *goja.Runtime, state *common.State,
 		tags["status"] = strconv.Itoa(resp.Status)
 		tags["proto"] = resp.Proto
 
+		if res.TLS != nil {
+			tlsState := res.TLS
+			switch tlsState.Version {
+			case tls.VersionSSL30:
+				resp.TLSVersion = "ssl3.0"
+			case tls.VersionTLS10:
+				resp.TLSVersion = "tls1.0"
+			case tls.VersionTLS11:
+				resp.TLSVersion = "tls1.1"
+			case tls.VersionTLS12:
+				resp.TLSVersion = "tls1.2"
+			}
+			resp.TLSCipherSuite = lib.SupportedTLSCipherSuitesToString[tlsState.CipherSuite]
+			resp.OCSP.StapledResponse = OCSPStapledResponse{}
+			if ocspRes, err := ocsp.ParseResponse(tlsState.OCSPResponse, nil); err == nil {
+				switch ocspRes.Status {
+				case ocsp.Good:
+					resp.OCSP.StapledResponse.Status = "good"
+				case ocsp.Revoked:
+					resp.OCSP.StapledResponse.Status = "revoked"
+				case ocsp.Unknown:
+					resp.OCSP.StapledResponse.Status = "unknown"
+				case ocsp.ServerFailed:
+					resp.OCSP.StapledResponse.Status = "server_failed"
+				}
+				switch ocspRes.RevocationReason {
+				case ocsp.Unspecified:
+					resp.OCSP.StapledResponse.RevocationReason = "unspecified"
+				case ocsp.KeyCompromise:
+					resp.OCSP.StapledResponse.RevocationReason = "key_compromise"
+				case ocsp.CACompromise:
+					resp.OCSP.StapledResponse.RevocationReason = "ca_compromise"
+				case ocsp.AffiliationChanged:
+					resp.OCSP.StapledResponse.RevocationReason = "affiliation_changed"
+				case ocsp.Superseded:
+					resp.OCSP.StapledResponse.RevocationReason = "superseded"
+				case ocsp.CessationOfOperation:
+					resp.OCSP.StapledResponse.RevocationReason = "cessation_of_operation"
+				case ocsp.CertificateHold:
+					resp.OCSP.StapledResponse.RevocationReason = "certificate_hold"
+				case ocsp.RemoveFromCRL:
+					resp.OCSP.StapledResponse.RevocationReason = "remove_from_crl"
+				case ocsp.PrivilegeWithdrawn:
+					resp.OCSP.StapledResponse.RevocationReason = "privilege_withdrawn"
+				case ocsp.AACompromise:
+					resp.OCSP.StapledResponse.RevocationReason = "aa_compromise"
+				}
+				resp.OCSP.StapledResponse.ProducedAt = ocspRes.ProducedAt.String()
+				resp.OCSP.StapledResponse.ThisUpdate = ocspRes.ThisUpdate.String()
+				resp.OCSP.StapledResponse.NextUpdate = ocspRes.NextUpdate.String()
+				resp.OCSP.StapledResponse.RevokedAt = ocspRes.RevokedAt.String()
```

Comment:
This should be in a format we can compare on the JS side... UNIX timestamps?

(I'm also kind of questioning what purpose it will serve for a load test script; I suppose it could be useful for logging, but not much more.)

Reply:
Yeah, should we use JS Date objects or Epoch milliseconds (to compare with `Date.now()` and `Date.getTime()`) when exposing to JS? Had the same question in the cookie PR.

Reply:
Just in the interest of performance, I think maybe giving a UNIX timestamp would be better than instancing whole date objects every time. The JS-native way seems to be to deal with milliseconds for them, so then you can just do `new Date(revokedAt)`.